### DISCUSSION TITLE
How to get user info?

### DISCUSSION THREAD

**Discussion by always-maap** [nextauthjs/next-auth#3526] [Help] [answered]
### Question 💬

calling user info endpoint directly([identity server 4](https://demo.identityserver.io/.well-known/openid-configuration) or [google](https://accounts.google.com/.well-known/openid-configuration)) with the access token is giving me the correct user info but how am I supposed to get it with next-auth?

```typescript
// user info example
{
  name: 'name',
  picture: 'picture',
  sub: '8ec2516a-ee0a-4f7a-80f0-77476752f6a1'
}
```
I've already read all the documentation and issues. 

### How to reproduce ☕️

```typescript
export default NextAuth({
  providers: [
    IdentityServer4Provider({
      id: 'identity-server4',
      name: 'IdentityServer4',
      issuer: 'https://demo.identityserver.io',
      clientId: 'interactive.public',
      clientSecret: 'secret',
      authorization: { params: { scope: 'openid profile' } },
    }),
  ]
});
```

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by always-maap** [comment]
@balazsorban44
I'm a bit confused. using `profileUrl` or `userinfo` should pass an arg to session callback. right? and then attaching to the session itself should do the job.
I could not find any example of that. even a small clue would be appreciated a lot

**Reply by balazsorban44** [reply]
there is no profileUrl anymore (gone in v4) . for userinfo, please see the docs

**Reply by always-maap** [reply]
that one sentence was enough. Thank you <3. maybe add Typescript helper to remind it is deprecated. I can make a pr

**Reply by balazsorban44** [reply]
it is gone form the types. there is no better helper IMO. 😅 also documented in the migration guide.

**Reply by always-maap** [reply]
I was confused by this. by yeah as far as the migration guide, you are right <3
![2021-12-28-162756_1920x1080_scrot](https://user-images.githubusercontent.com/58308349/147568843-5a8bef74-3022-4fa1-a0a3-eff9b9c8672c.png)

**Reply by balazsorban44** [reply]
hmm. apologies, it sounds like a leftover then. 😬 I'll have a look if I don't forget. 😂 otherwise feel free to address it in a PR. You could check the source code if  we don't accidentally rely on it in some weird way

**Comment by always-maap** [comment, marked as answer]
working code for custom user info. identity server 4
```typescript
import axios from 'axios';
import NextAuth from 'next-auth';
import IdentityServer4Provider from 'next-auth/providers/identity-server4';

export default NextAuth({
  providers: [
    IdentityServer4Provider({
      id: 'identity-server4',
      name: 'IdentityServer4',
      issuer: 'https://demo.identityserver.io',
      clientId: 'interactive.public',
      clientSecret: 'secret',
      userinfo: {
        url: 'https://demo.identityserver.io/connect/userinfo',
        async request(context) {
          const res = await axios.get('https://demo.identityserver.io/connect/userinfo', {
            headers: {
              Authorization: `Bearer ${context.tokens.access_token}`,
            },
          });
          return res.data;
        },
      },
      profile: (profile) => {
        return {
          id: profile.sub,
          name: profile.name,
          picture: profile.picture,
        };
      },
      authorization: { params: { scope: 'openid profile' } },
      idToken: true,
    }),
  ],
  callbacks: {
    async session({ session, user, token }) {
      session.user = token.user;

      return session;
    },
    async jwt({ token, user, account, profile, isNewUser }) {
      if (profile) {
        token.user = profile;
      }

      return token;
    },
  },
});

```
